### PR TITLE
[reboot] Add a power_off_delay option for reboot test to allow customizing the power off delay

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
### Description of PR

Summary:
Add a power_off_delay option for reboot test to allow customizing the power off delay

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?

Currenty test_power_off_reboot case use 5 and 15 as power off delay by default, the power delay seconds are hard coded in pytest fixture. This PR allows user to customize the power off delay by pytest option. 

#### How did you do it?

Add a power_off_delay option. If the option is not specified, it will use default [5, 15] value by default. The option can be specified multi times. For example: "--power_off_delay=8 --power_off_delay=9 --power_off_delay=10", test_power_off_reboot will run with power off delay 8, 9 and 10. 

#### How did you verify/test it?

Manual test

#### Any platform specific information?

N/A

#### Supported testbed topology if it's a new test case?

N/A

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -135,6 +135,17 @@ def pytest_addoption(parser):
...
+    #   reboot test options #
+    #########################
+
+    # Allow user to specify the power off delay in seconds. User can add multiple delay by:
+    # "--power_off_delay=5, --power_off_delay=15, --power_off_delay=20"
+    # If this option is not specified, [5, 15] will be used by default, see function pytest_generate_tests in
+    # tests/platform_tests/conftest.py
+    parser.addoption("--power_off_delay", action="append", type=int, default=[],
+                     help="Power off delay seconds for test_power_off_reboot")
+
```
[the remark this answers is not in the document]
Comment (Junchao-Mellanox) [reply tests/conftest.py:148]: Thanks Xin. I will have a test on that.

[the remark this answers is not in the document]
Comment (Junchao-Mellanox) [reply tests/conftest.py]: Moved power_off_delay option to platform_tests/conftest.py

**Review by wangxin — APPROVED** [review]

File: tests/conftest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (wangxin) [reply tests/conftest.py]: Thanks!

[the remark this answers is not in the document]
Comment (wangxin) [reply tests/conftest.py]: BTW, the reason of why options defined in subfolder `conftest.py` may be unrecognized is explained here: https://github.com/Azure/sonic-mgmt/pull/2131

[the remark this answers is not in the document]
Comment (Junchao-Mellanox) [reply tests/conftest.py]: Many thanks to that!

**Pull request merged**